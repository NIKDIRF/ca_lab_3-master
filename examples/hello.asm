.data:
    string hw "Hello world!"
    num cur_ind 0
.text:
    mov [hw] cur_ind
loop: out# cur_ind
    cmp* cur_ind 0
    je end
    jmp loop
end: hlt


