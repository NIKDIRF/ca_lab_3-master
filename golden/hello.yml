source: |
  .data:
      string hw "Hello world!"
      num cur_ind 0
  .text:
      mov [hw] cur_ind
  loop: out# cur_ind
      cmp* cur_ind 0
      je end
      jmp loop
  end: hlt
data:
- 72
- 101
- 108
- 108
- 111
- 32
- 119
- 111
- 114
- 108
- 100
- 33
- 0
- '0'
- 0
- '0'
code:
- opcode: mov
  addr: 0
  args:
  - 14
  - 13
- opcode: out_rel
  addr: 1
  args:
  - 13
- opcode: cmp_rel_inc
  addr: 2
  args:
  - 13
  - 15
- opcode: je
  addr: 3
  args:
  - 5
- opcode: jmp
  addr: 4
  args:
  - 1
- opcode: hlt
  addr: 5
  args: []
output: "Hello world!\0"
